gluten-arrow/src/main/java/org/apache/gluten/vectorized/ColumnarBatchOutIterator.java

```diff
@@ -54,6 +54,9 @@ public long itrHandle() {
 
   @Override
   public boolean hasNext0() throws IOException {
+    if (closed.get()) {
+      return false;
+    }
```

Comment:
Not meant to be nitpicking, but I think this is not needed for now?

`hasNext() == false` literally means the iterator was drained internally while `closed == true` could mean the iterator was closed by a caller. The two are semantically different so I think we can add this kind of logic when it's really needed.

Reply:
Makes sense, I will remove this, thank you.

Reply:
But after closed, the native iterator has been removed. Does this mean that this iterator was drained?